sqlglot/parser.py

```diff
@@ -6292,3 +6298,94 @@ def _parse_with_operator(self) -> t.Optional[exp.Expression]:
         op = self._parse_var(any_token=True)
 
         return self.expression(exp.WithOperator, this=this, op=op)
+
+    def _parse_copy_parameters(self) -> t.List[exp.CopyParameter]:
+        sep = self.dialect.COPY_PARAMS_SEP
+
+        options = []
+        while self._curr and not self._match(TokenType.R_PAREN, advance=False):
+            option = self._parse_unquoted_field()
+            value = None
+            # Some options are defined as functions with the values as params
+            if not isinstance(option, exp.Func):
+                # Different dialects might separate options and values by white space, "=" and "AS"
+                eq = self._match(TokenType.EQ)
+                self._match(TokenType.ALIAS)
+                value = self._parse_unquoted_field()
+                if eq:
+                    option = exp.EQ(this=option, expression=value)
+                    value = None
+
+            param = self.expression(exp.CopyParameter, this=option, value=value)
+            options.append(param)
+
+            if sep:
+                self._match(sep)
+
+        return options
+
+    def _parse_credentials(self) -> t.Optional[exp.Credentials]:
+        def parse_options():
+            opts = []
+            self._match(TokenType.EQ)
+            self._match(TokenType.L_PAREN)
+            while not self._match(TokenType.R_PAREN):
+                opts.append(self._parse_conjunction())
+            return opts
+
+        expr = self.expression(exp.Credentials)
+
+        if self._match_text_seq("STORAGE_INTEGRATION", advance=False):
+            expr.set("storage", self._parse_conjunction())
+        if self._match_text_seq("CREDENTIALS"):
+            # Snowflake supports CREDENTIALS = (...), while Redshift CREDENTIALS <string>
+            creds = parse_options() if self._match(TokenType.EQ) else self._parse_field()
+            expr.set("credentials", creds)
+        if self._match_text_seq("ENCRYPTION"):
+            expr.set("encryption", parse_options())
+        if self._match_text_seq("IAM_ROLE"):
+            expr.set("iam_role", self._parse_field())
+        if self._match_text_seq("REGION"):
+            expr.set("region", self._parse_field())
+
+        return expr
+
+    def _parse_copy(self):
+        start = self._prev
+
+        into = self._match_text_seq("INTO")
+
+        this = (
+            self._parse_conjunction()
+            if self._match(TokenType.L_PAREN, advance=False)
+            else self._parse_table(schema=True)
+        )
+
+        if self._match(TokenType.FROM):
+            kind = True
+        elif self._match_text_seq("TO"):
+            kind = False
```

Comment:
We need to set a default value to `kind`, we'll get a `NameError` instead of a `ParseError`